tools/update-locales.go

```diff
@@ -0,0 +1,113 @@
+// Copyright 2025 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build ignore
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+const (
+	localeDir       = "options/locale"
+	defaultLocaleFileName  = "locale_en-US.json"
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	// remove empty values
+	localeFiles, err := filepath.Glob(filepath.Join(localeDir, "*.json"))
+	if err != nil {
+		return fmt.Errorf("list %s: %w", localeDir, err)
+	}
+
+	for _, file := range localeFiles {
+		if err := cleanLocaleFile(file); err != nil {
+			return fmt.Errorf("clean %s: %w", file, err)
+		}
+	}
+
+	// remove incomplete translations
+	originalLocalePath := filepath.Join(localeDir, defaultLocaleFileName)
+	baselineLines, err := countLines(originalLocalePath)
+	if err != nil {
+		return fmt.Errorf("count %s: %w", originalLocalePath, err)
+	}
+	threshold := baselineLines / 4 // 25% of baseline
+
+	for _, file := range localeFiles {
+		lines, err := countLines(file)
+		if err != nil {
+			return fmt.Errorf("count %s: %w", file, err)
+		}
+		if lines < threshold {
+			fmt.Printf("Removing %s: %d/%d\n", file, lines, threshold)
+			if err := os.Remove(file); err != nil {
+				return fmt.Errorf("remove %s: %w", file, err)
+			}
+		}
```

Comment:
`Removing options/locale/locale_ar-SA.json: 64/947` reads really strangely. What's 64/947? why is the file removed?
Maybe:
```suggestion
    fmt.Printf("Removing locale files with less than %d non-empty keys\n", threshold)
		if lines < threshold {
      fmt.Printf("%s: %d\n", file, lines)
			if err := os.Remove(file); err != nil {
				return fmt.Errorf("remove %s: %w", file, err)
			}
		}
```
?
That way at least the "why" and "what" is clear